src/main/java/io/epirus/console/deploy/DeployRunner.java

```
@@ -83,50 +95,37 @@ public void deploy() throws Exception {
-                            credentials.getAddress(),
-                            Faucet.valueOf(network.getNetworkName().toUpperCase()),
-                            "alex");
-                }
-                return;
-            } catch (Exception e) {
-
-                try {
-                    Thread.sleep(2000);
-                } catch (InterruptedException ex) {
-                    Thread.currentThread().interrupt();
-                }
+        try {
+            BigInteger accountBalance =
+                    accountManager.getAccountBalance(credentials, network, web3j);
+            if (accountBalance.equals(BigInteger.ZERO)) {
+                WalletFunder.fundWallet(
+                        credentials.getAddress(),
+                        Faucet.valueOf(network.getNetworkName().toUpperCase()),
+                        this.accountManager.getLoginToken());
             }
+        } catch (Exception e) {
+            Console.exitError("Could not fund wallet: " + e.getMessage());
         }
     }
 
     private void waitForBalanceUpdate() {
         System.out.println("Checking the account balance...");
-        int tries = 5;
-        while (tries-- > 0) {
-            try {
-                BigInteger accountBalance = accountManager.getAccountBalance(credentials, network);
-                if (!accountBalance.equals(BigInteger.ZERO)) {
-                    return;
-                } else {
-                    Thread.sleep(10000);
-                }
-            } catch (Exception e) {
-                try {
-                    Thread.sleep(2000);
-                } catch (InterruptedException ex) {
-                    Thread.currentThread().interrupt();
-                }
-            }
+        try {
+            BigInteger accountBalance =
+                    accountManager.pollForAccountBalance(credentials, network, web3j, 5);
+
+        } catch (Exception e) {
+            e.printStackTrace();
```

Swallowing exception, should do exitError instead